.github/actions/install-libs/action.yml

```diff
@@ -0,0 +1,39 @@
+name: Install Cylc libs
+description: Install cylc-flow and cylc-rose at correct (or specified) versions.
+
+inputs:
+  cylc-flow-repo:
+    description: The cylc-flow GitHub repo/fork to install
+    required: false
+  cylc-flow-ref:
+    description: cylc-flow branch/tag/etc
+    required: false
+  cylc-rose-ref:
+    description: cylc-rose branch/tag etc
+    required: false
+
+
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      env:
+        CYLC_FLOW_GITHUB: git+https://github.com/${{ inputs.cylc-flow-repo || 'cylc/cylc-flow' }}
+        CYLC_FLOW_REF: ${{ inputs.cylc-flow-ref }}
+        CYLC_ROSE_GITHUB: git+https://github.com/cylc/cylc-rose
+        CYLC_ROSE_REF: ${{ inputs.cylc-rose-ref }}
+      run: |
+        # NOTE: hard-coded branches need to be manually updated as necessary.
+        if [[ "$GITHUB_BASE_REF" == '2.0.x' ]]; then
+          : ${CYLC_FLOW_REF:='8.0.x'}
+          : ${CYLC_ROSE_REF:='1.1.x'}
```

Comment:
This syntax assigns to the env vars if they are not set
https://stackoverflow.com/a/28085062/3217306